Tools/cases_generator/stacking.py

```diff
@@ -202,11 +225,27 @@ def __init__(
             else:
                 pred = None  # Break
 
+        # Fix up patterns of copies through UNUSED,
+        # e.g. cp(a, UNUSED) + cp(UNUSED, b) -> cp(a, b).
+        if any(copy.src.effect.name == UNUSED for copy in self.copies):
+            pred = self.pred
+            while pred is not None:
+                for copy in self.copies:
+                    if copy.src.effect.name == UNUSED:
```

Comment:
```suggestion
        unused_copies = (copy for copy in self.copied if copy.src.effect.name == UNUSED)
        if unused_copies:
            pred = self.pred
            while pred is not None:
                for copy in unused_copies:
```

Reply:
I'm not so sure about this suggestion. The innermost loop actually modifies the set of copies whose source is an 'unused' variable. I'll keep what I had.